submissions/Roka20012/js-oop/app.js

```diff
@@ -0,0 +1,147 @@
+        this.x = x;
+        this.y = y;
+        this.sprite = sprite;
+    }
+
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+}
+
+class Player extends Heroes {
+    constructor(x, y, sprite = "images/char-boy.png") {
+        super(x, y, sprite);
+    }
+
+    update(dt) {
+        switch (this.x) {
+            case PLAYER_CONF.MAX_X_RIGHT:
+                this.x = PLAYER_CONF.MAX_X_RIGHT - PLAYER_CONF.MOVE_LEFT_RIGTH;
+                break;
+            case PLAYER_CONF.MAX_X_LEFT:
+                this.x = PLAYER_CONF.MAX_X_LEFT + PLAYER_CONF.MOVE_LEFT_RIGTH;
+                break;
+        }
+
+        switch (this.y) {
+            case PLAYER_CONF.MAX_Y_UP:
+                score += 1;
+                scoreBoard.innerHTML = `Max Score:${maxScore} Score:${score}`;
+                this.x = PLAYER_CONF.START_X;
+                this.y = PLAYER_CONF.START_Y;
+                break;
+            case PLAYER_CONF.MAX_Y_DOWN:
+                this.y = PLAYER_CONF.START_Y;
+        }
+    }
+
+    handleInput(key) {
+        switch (key) {
+            case 'up':
+                this.y -= PLAYER_CONF.MOVE_UP_DOWN;
+                break;
+            case 'down':
+                this.y += PLAYER_CONF.MOVE_UP_DOWN;
+                break;
+            case 'left':
+                this.x -= PLAYER_CONF.MOVE_LEFT_RIGTH;
+                break;
+            case 'right':
+                this.x += PLAYER_CONF.MOVE_LEFT_RIGTH;
+                break;
+        }
+    }
+
+}
+
+class Enemy extends Heroes {
+    constructor(x, y, speed, sprite = 'images/enemy-bug.png') {
+        super(x, y, sprite);
+        this.speed = speed;
+    }
+
+    update(dt) {
+        if (this.x <= ENEMY_CONF.MAX_X) {
+            this.x += this.speed;
+        } else {
+            this.x = ENEMY_CONF.START_X;
+            this.speed = getRandomSpeed(ENEMY_CONF.MIN_SPEED + score, ENEMY_CONF.MAX_SPEED + score);
+        }
+
+        if (player.x < this.x + ENEMY_CONF.CHECK_X_Y &&
+            player.x + ENEMY_CONF.CHECK_X_Y > this.x &&
+            player.y < this.y + ENEMY_CONF.CHECK_X_Y &&
+            player.y + ENEMY_CONF.CHECK_X_Y > this.y) {
+
+            if (maxScore < score) maxScore = score;
+            score = 0;
+            scoreBoard.innerHTML = `Max Score:${maxScore}  Score:${score}`;
+            player.x = PLAYER_CONF.START_X;
+            player.y = PLAYER_CONF.START_Y;
+        }
+    }
+}
+
+function getRandomSpeed(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+
+    return Math.floor(Math.random() * (max - min)) + min;
+}
+
+
+
+
+let player = new Player(PLAYER_CONF.START_X, PLAYER_CONF.START_Y);
+
+const allEnemies = [];
+
+function newEnemy(y = 0, player) {
+    let enemy = new Enemy(ENEMY_CONF.START_X, ENEMY_CONF.START_Y + y,
+        getRandomSpeed(ENEMY_CONF.MIN_SPEED, ENEMY_CONF.MAX_SPEED));
+    allEnemies.push(enemy);
+}
+
+newEnemy();
+newEnemy(ENEMY_CONF.NEXT_Y);
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and how about `allEnemies = [someParameter1, someParameter2, someParameter3].map(position => new Enemy(ENEMY_CONF.START_X, ENEMY_CONF.START_Y + position ... ))`

Reply:
Thanks for the advice)Fixed)